src/rabbit_queue_consumers.erl

```diff
@@ -270,6 +280,57 @@ subtract_acks(ChPid, AckTags, State) ->
             end
     end.
 
+% subtract_acks_impl removes the acks specified in the first parameter,
+% the second and third paramters should be empty array and empty map
+% the fourth paramter is the ack queue
+% it returns {map(CTag => Count), AckQueue}
+subtract_acks_impl([], CTagCounts, AckQ) ->
+    {CTagCounts, AckQ};
+subtract_acks_impl([T | TL], CTagCounts, AckQ) ->
+    case maps:take(T, AckQ) of
+      {CTag, AckQ1} ->
+        subtract_acks_impl(TL, maps:update_with(CTag, fun (Old) -> Old + 1 end, 1, CTagCounts), AckQ1);
```

Comment:
Do we need `maps:update_with/3` here? Looks like we can safe a function call by using `maps:update/3` since we're only incrementing a counter.